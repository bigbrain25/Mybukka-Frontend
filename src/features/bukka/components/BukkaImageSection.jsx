import React from 'react';
import PropTypes from 'prop-types';

import './bukkaImage.scss';

const BukkaImage = ({ imageUrl }) => (
  <div
    className="bukka-image-section"
    style={{ backgroundImage: `url(${imageUrl})` }}
  />
);

export default BukkaImage;

BukkaImage.propTypes = {
  imageUrl: PropTypes.string.isRequired,
};
