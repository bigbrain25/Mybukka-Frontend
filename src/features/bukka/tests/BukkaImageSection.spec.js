import React from 'react';
import BukkaImageSection from '../components/BukkaImageSection';

describe('BukkaImageSection Component', () => {
  const props = {
    imageUrl: 'http://some-img-url',
  };

  const wrapper = mount(<BukkaImageSection {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
