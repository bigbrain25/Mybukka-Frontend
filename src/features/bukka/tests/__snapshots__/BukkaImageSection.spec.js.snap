// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BukkaImageSection Component renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <BukkaImage
    imageUrl="http://some-img-url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "imageUrl": "http://some-img-url",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="bukka-image-section"
        style="background-image: url(http://some-img-url);"
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "bukka-image-section",
        "style": Object {
          "backgroundImage": "url(http://some-img-url)",
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "imageUrl": "http://some-img-url",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="bukka-image-section"
          style="background-image: url(http://some-img-url);"
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bukka-image-section",
          "style": Object {
            "backgroundImage": "url(http://some-img-url)",
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
